src/AuthCredentials.js

```diff
@@ -0,0 +1,88 @@
+import Credentials from './Credentials.js'
+import Storage from './Storage.js'
+import crypto from 'crypto'
+import { createJWT } from './JWT'
+
+// Generate random 16 byte hex string
+const random  = () => {
+  return new Promise((resolve, reject) => {
+    crypto.randomBytes(16, (err, buf) => {
+      if (err) reject(error)
+      resolve(buf.toString('hex'))
+    })
+  })
+}
+
+class AuthCredentials extends Credentials {
+  constructor (settings = {}) {
+    super(settings)
+    this.challengeStorage = settings.challengeStorage || new Storage('challenge')
+    this.responseStorage = settings.responseStorage || new Storage('response')
```

Comment:
AuthCredentials should not use responseStorage. This makes too many assumptions about the implementation and integration into the consuming applications architecture.